rs/sns/treasury_manager/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+use candid::{CandidType, Nat, Principal};
+use serde::{Deserialize, Serialize, Serializer};
+use std::{collections::BTreeMap, fmt::Display};
+
+#[derive(CandidType, Clone, Debug, Deserialize)]
+pub struct TreasuryManagerInit {
+    pub allowances: Vec<Allowance>,
+}
+
+#[derive(CandidType, Clone, Debug, Deserialize)]
+pub struct TreasuryManagerUpgrade {}
+
+#[derive(CandidType, Clone, Debug, Deserialize)]
+pub enum TreasuryManagerArg {
+    Init(TreasuryManagerInit),
+    Upgrade(TreasuryManagerUpgrade),
+}
+
+#[derive(CandidType, Clone, Debug, Deserialize, PartialEq)]
+pub struct Balance {
+    #[serde(serialize_with = "serialize_nat_as_u64")]
+    pub amount_decimals: Nat,
+    pub account: Option<Account>,
+}
+
+#[derive(CandidType, Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
+pub enum Party {
+    TreasuryOwner,
+    TreasuryManager,
+    External,
+    LedgerFee,
```

Comment:
I don't understand this "Party" concept... especially since one of the entries if "LedgerFee".